crates/assists/src/utils/insert_use.rs

```diff
@@ -22,525 +18,566 @@ pub(crate) fn find_insert_use_container(
 ) -> Option<Either<ast::ItemList, ast::SourceFile>> {
     ctx.sema.ancestors_with_macros(position.clone()).find_map(|n| {
         if let Some(module) = ast::Module::cast(n.clone()) {
-            return module.item_list().map(|it| Either::Left(it));
+            return module.item_list().map(Either::Left);
         }
         Some(Either::Right(ast::SourceFile::cast(n)?))
     })
 }
 
-/// Creates and inserts a use statement for the given path to import.
-/// The use statement is inserted in the scope most appropriate to the
-/// the cursor position given, additionally merged with the existing use imports.
 pub(crate) fn insert_use_statement(
     // Ideally the position of the cursor, used to
     position: &SyntaxNode,
     path_to_import: &str,
-    ctx: &AssistContext,
-    builder: &mut TextEditBuilder,
+    ctx: &crate::assist_context::AssistContext,
+    builder: &mut text_edit::TextEditBuilder,
 ) {
-    let target = path_to_import.split("::").map(SmolStr::new).collect::<Vec<_>>();
-    let container = find_insert_use_container(position, ctx);
+    insert_use(position.clone(), make::path_from_text(path_to_import), Some(MergeBehaviour::Full));
+}
+
+pub fn insert_use(
+    where_: SyntaxNode,
+    path: ast::Path,
+    merge_behaviour: Option<MergeBehaviour>,
```

Comment:
```suggestion
    merge: Option<MergeBehaviour>,
```

I took me a while to get why this needs to be an `Option`, maybe renaming to `merge` help?

Reply:
That's a good point, figured that the `Option` wrapping might be a bit confusing, changed the name and added a small doc comment.